cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/planner/Dag.java

```diff
@@ -138,171 +136,39 @@ void validate() {
   }
 
   /**
-   * Split the dag based on the input control nodes.
-   * @param controlNodes set of conditions and actions based on which to split the Dag
+   * Split the dag based on the input control nodes. Control nodes represent actions or conditions.
+   * Actions can only be at the front or end of the dag. Condition nodes can be anywhere in the dag, but can only
+   * have at most 2 outputs. Additionally, there must not be any intersection between the nodes accessible its
```

Comment:
nit - the last sentence in this paragraph isn't clear to me
do you mean "between the nodes accessible by its outputs"?

Reply:
Is it clearer if it is:

`have at most 2 output branches. Output branches must never intersect.`

Trying to same that the condition branches should never have any nodes in common.

Reply:
yea, that's simpler and makes sense.